src/processTargets/modifiers/OrdinalStages.ts

```diff
@@ -0,0 +1,112 @@
+import { Target } from "../../typings/target.types";
+import {
+  AbsoluteOrdinalScopeModifier,
+  RelativeOrdinalScopeModifier,
+  ScopeType,
+} from "../../typings/targetDescriptor.types";
+import { ProcessedTargetsContext } from "../../typings/Types";
+import getModifierStage from "../getModifierStage";
+import { ModifierStage } from "../PipelineStages.types";
+
+export class AbsoluteOrdinalStage implements ModifierStage {
+  constructor(private modifier: AbsoluteOrdinalScopeModifier) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    let targets = getTargets(context, target, this.modifier.scopeType);
+
+    // For explicit ranges use that range as the iteration scope
+    if (target.hasExplicitRange) {
+      targets = targets.filter((t) => {
+        const intersection = t.contentRange.intersection(target.contentRange);
+        return intersection != null && !intersection.isEmpty;
+      });
+    }
+
+    const startIndex =
+      this.modifier.start + (this.modifier.start < 0 ? targets.length : 0);
+    const endIndex = startIndex + this.modifier.length - 1;
+
+    return [createTarget(target, targets, startIndex, endIndex)];
+  }
+}
+
+export class RelativeOrdinalStage implements ModifierStage {
+  constructor(private modifier: RelativeOrdinalScopeModifier) {}
+
+  run(context: ProcessedTargetsContext, target: Target): Target[] {
+    const targets = getTargets(context, target, this.modifier.scopeType);
+    const isForward = this.modifier.direction === "forward";
+
+    const containingIndices = targets
+      .map((t, i) => ({
+        index: i,
+        intersection: t.contentRange.intersection(target.contentRange),
+      }))
+      .filter((t) => t.intersection != null)
```

Comment:
In case of an empty selection we would get no containing indices. ie empty selection in the middle of a function will give no result which is not what we want.